docs/content/release-notes/0.3.0.md

```diff
@@ -0,0 +1,157 @@
+---
+title: "0.3.0"
+date: "2016-10-20"
+slug: "0-3-0"
+
+menu:
+  main:
+    parent: "release-notes"
+    identifier: 0.3.0
+
+---
+
+## Release Date
+
+2016-10-21 0.3.0-beta1
+
+## Features
+
+This release includes a number of significant improvements to the core engine including conditionals,
+wait queries, sequential task groups, and the ability to use lists and maps in parameters. The internal API has
+also been simplified to make it easier for programmers to create new modules.
+
+### Module Improvements
+
+#### RPM Resource
+
+RPM package install/uninstall support has been added via [#373](https://github.com/asteris-llc/converge/pull/373).
+
+```hcl
+rpm.package "mc" {
+    name  = "mc"
+    state = "present"
+}
+```
+
+#### User and group
+
+Linux user groups can now be modified via [#279](https://github.com/asteris-llc/converge/issues/279)
+
+#### wait.query and wait.port
+
+Converge now supports waiting for a task to complete or port [#334](https://github.com/asteris-llc/converge/pull/334).
+Users can create a dependency for a port to be open or a condition to be met.
+
+Below is an example of waiting for a port to be open using the `wait.port` resource:
+
+```hcl
+wait.port "8080" {
+  host         = "localhost"
+  port         = 8080
+  interval     = "1s"
+  max_retry    = 10
+  grace_period = "2s"
+}
+```
+
+The following shows using `wait.query` to wait using a task:
+
+```hcl
+wait.query "service-health" {
+  check        = "nc -z localhost 8080"
+  interval     = "1s"
+  max_retry    = 10
+  grace_period = "1s"
+}
+```
+
+Wait is documented at: [wait.port]({{< relref "resources/wait.port.md" >}}) and [wait.query]({{< relref "resources/wait.query.md" >}}).
+
+
+### Engine Improvements
+
+#### Conditional Support
+
+Support for conditionals via [#362](https://github.com/asteris-llc/converge/pull/362).
+
+In the following example, the contents of file `greetings.txt` depend on the value of
+the `lang` parameter:
+
+```hcl
+param "lang" {
+  default = ""
+}
+
+switch "test-switch" {
+  case "eq `spanish` `{{param `lang`}}`" "spanish" {
+    file.content "foo-file" {
+      destination = "greeting.txt"
+      content     = "hola\n"
+    }
+  }
+
+  case "eq `french` `{{param `lang`}}`" "french" {
+    file.content "foo-file" {
+      destination = "greeting.txt"
+      content     = "salut\n"
+    }
+  }
+
+```
+
+#### Task Groups
+
+Tasks can be assigned to a group to force multiple tasks to run
+one at a time instead of in parallel. Locks are added as a node in the graph and released
```

Comment:
this is slightly out of date with the latest implementation. locks are no longer nodes in the graph; instead we just ensure all nodes in the group are dependent on another node in the group.